assets/src/edit-story/app/prepublish/checklistProvider.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../story';
+import Context from './context';
+import prepublishChecklist from '.';
+
+function ChecklistProvider({ children }) {
+  const { story, pages } = useStory(({ state: { story, pages } }) => ({
+    story,
+    pages,
+  }));
+
+  const [currentList, setCurrentList] = useState(
+    prepublishChecklist({ ...story, pages })
+  );
+
+  useEffect(() => {
+    setCurrentList(prepublishChecklist({ ...story, pages }));
+  }, [story, pages]);
```

Comment:
Does that potentially run very often?

Reply:
Also here is where I am assuming @swissspidy would want the `prepublishChecklist` function to turn into `usePrepublishChecklist`.

Reply:
Ok, I wasn't sure what would be the most user-friendly but not bogging down the app is probably the first trade off to make for the first increment. I think adding a manual check would probably help, as well as running when preview/publish is clicked. What I mean by "manual check" was I was thinking of including a sort of "refresh" button on the tab itself, in case the tab is active so it's not confusing about why, for example, "missing title" might still be there even though you write a title in the box. 

I was also thinking it could be included as part of the `autosave` functionality, but I think these are all good discussions to have for future improvements. I'll refactor the hook to start.